### DISCUSSION TITLE
Github Copilot Chat - cannot access full file context

### DISCUSSION THREAD

**Discussion by rushiagr** [community/community#61801] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

The [official docs](https://docs.github.com/en/copilot/github-copilot-chat/using-github-copilot-chat#asking-github-copilot-chat-questions-about-your-code) provides an example question `What does this file do?`, but when I ask the same in github copilot chat section in visual studio code, it just talks about the code which is visible on the screen. This is very limiting and not sure how to ask it to read the whole file's context (i'm not even talking about the whole project yet). 

If this is not possible, then we should update docs to highlight it

**Comment by samus-aran** [comment]
Hi @rushiagr 👋 

Thanks for this feedback, can I ask for more clarity when you say..

> it just talks about the code which is visible on the screen

Are you meaning that it is not including all the code that resides in the file you have open, only the code that is visible within your screen's resolution (on Visual Studio)?

**Comment by rushiagr** [comment]
Yes, that is correct. The file is more than 1000 lines, and copilot only referenced code which is visible on the screen (within the screen resolution)

**Comment by Akash1134** [comment]
Hi @rushiagr ,

Thank you for sharing the feedback. I have shared it with my team internally. 🙂 

We will update you once we have more details, clarification, or further questions.

**Reply by saxenaprerit** [reply]
Any update on this? I still see this issue.

**Reply by diarmaidocualain** [reply]
Same here. If I as "Could you explain what this file does please? ", it will only explain what the code does that is displayed on the screen, not all of the file. 

Workaround: If I highlight all the code in the file and as "What does this code do", it will explain all the code in the file.

**Reply by log101** [reply]
Any updates? Also can you clarify if this is a feature or a bug?

**Comment by digitarald** [comment]
This tracked here: https://github.com/microsoft/vscode-copilot-release/issues/173

**Comment by clothy-terry** [comment]
You can try to select all lines of code you want it to read, and then type you question to chat. It will generate the response base on your selected(NOT copy&paste) content. But it might fail when there are too many lines of code.

**Reply by ghost** [reply]
Can I add to this discussion?

I suddenly have the same/similar problem. Yesterday CP chat worked on a large bash script I had open in VS Code. This morning CP chat claims it cannot access the file, the same file it worked on yesterday evening. If I copy the full script into the chat window it does work but this is a pretty annoying step that suddenly emerged.

**Reply by christopher-gomez** [reply]
Same thing happening to me. Saying it doesn't have any access to my files and only what I share in the context. That's simply not true and it should have access to the files in my VS code workspace

**Reply by btiernay** [reply]
Same for me. It cannot access any file content unless it is selected in the active editor.

**Reply by btiernay** [reply]
Created a new discussion here for inability to access files in Chat:
https://github.com/orgs/community/discussions/110209

**Comment by ayim** [comment, marked as answer]
You should be able to reference the entire file with the # command now.
See details here: https://github.com/orgs/community/discussions/102847
